src/compression/CompressionFactory.h

```diff
@@ -0,0 +1,23 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2017 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include "ICompression.h"
+#define IS_COMPRESSION_ENABLED 1
+
+typedef enum ZipMethod
+{
+    gzip,
+    rvl,
+    jpeg,
+    lz,
+} ZipMethod;
+
+class CompressionFactory
```

Comment:
Since compression factory is a singleton please change the design as follows:
a. All methods are not static, enabled is class member
b. Add single static function:
```
static CompressionFactory& CompressionFactory::instance() {
     static CompressionFactory inst;
     return inst;
}
```
aka mayers singleton.
also, consider not making it singleton

Reply:
WIP